File: typescript/compete/code_golf/dont-panic.ts
let r=()=>readline(),[,,,,a,,,b]=r().split(' ').map(i=>+i),c=[],d=[]
for(let i=0;i<b;i++){
let [e,f]=r().split(' ').map(i=>+i)
c.push(e)
d.push(f)}
for(;;){var g=r().split(' '),h=+g[1],j=g[2],k=c.indexOf(+g[0]),l=k>=0?d[k]:a
console.log(((l<h&&j[0]=='L')||(h<l&&j[0]=='R')||l==h)?'WAIT':'BLOCK')}

File: go/compete/code_golf/dont-panic.go
package main
import ."fmt"
func main(){
var a,b,c,d,e,f,z int
Scan(&z,&z,&z,&z,&a,&z,&z,&b)
var g string
h:=[]int{}
i:=[]int{}
for z=0;z<b;z++{Scan(&c,&d)
h=append(h,c)
i=append(i,d)}
for{Scan(&e,&f,&g)
q:=a
for z=0;z<len(h);z++{if h[z]==e{q=i[z]
break}}
t:="BLOCK"
if(q<f&&g=="LEFT")||(f<q&&g[0]=='R')||q==f{t="WAIT"}
Println(t)}}
